crates/red_knot_python_semantic/resources/mdtest/suppressions/no-type-check.md

```diff
@@ -0,0 +1,94 @@
+# `@no_type_check`
+
+> If a type checker supports the `no_type_check` decorator for functions, it should suppress all type errors for the def statement and its body including any nested functions or classes. It should also ignore all parameter and return type annotations and treat the function as if it were unannotated.
+> [source](https://typing.readthedocs.io/en/latest/spec/directives.html#no-type-check)
+
+## Error in the function body
+
+```py
+from typing import no_type_check
+
+@no_type_check
+def test() -> int: 
+    return a + 5
+```
+
+## Error in nested function
+
+```py
+from typing import no_type_check
+
+@no_type_check
+def test() -> int:
+    def nested(): 
+        return a + 5
+```
+
+## Error in nested class
+
+```py
+from typing import no_type_check
+
+@no_type_check
+def test() -> int:
+    class Nested:
+        def inner(self):
+            return a + 5
+```
+
+## Error in decorator
+
+Both MyPy and Pyright flag the `unknown_decorator` but we don't. 
+
+```py
+from typing import no_type_check
+
+@unknown_decorator
+@no_type_check
+def test() -> int:
+    return a + 5
+```
+
+## Error in default value
+
+```py
+from typing import no_type_check
+
+@no_type_check
+def test(a: int = "test"): 
+    return x + 5
+```
+
+## Error in return value position
+
+```py
+from typing import no_type_check
+
+@no_type_check
+def test() -> Undefined: 
+    return x + 5
+```
+
+## `no_type_check` on classes isn't supported
+
+Similar to Pyright, Red Knot does not support `no_type_check` annotations on classes.
+
+```py
+from typing import no_type_check
+
+@no_type_check
+class Test:
+    def test(self):
+        return a + 5  # error: [unresolved-reference]
+```
+
+## `type: ignore` comments in `@no_type_check` blocks
+
+```py
+from typing import no_type_check
+
+@no_type_check
+def test():
+    #  error: [unused-ignore-comment]
```

Comment:
```suggestion
    # error: [unused-ignore-comment]
```

Also, I wonder if it would be nice to have an additional explanation included in the error message here, such as:

```suggestion
    # error: [unused-ignore-comment] "Unused ignore comment (all errors in `test` are already ignored due because it is decorated with `@no_type_check`)"
```

Reply:
> Also, I wonder if it would be nice to have an additional explanation included in the error message here, such as:

Possibly, but it requires some extra metadata tracking for suppressions or at least post-processing in the `unused-ignore-comment` rule. I don't think it's a priority right now (we already go beyond by flagging unused ignore comments